core/src/main/c/share/net.c

```diff
@@ -321,3 +323,34 @@ JNIEXPORT jint JNICALL Java_io_questdb_network_Net_getPeerPort
     }
     return -1;
 }
+
+JNIEXPORT jint JNICALL Java_io_questdb_network_Net_connectAddrInfo
+        (JNIEnv *e, jclass cl, jlong fd, jlong lpAddrInfo) {
+    struct addrinfo *addr = (struct addrinfo *) lpAddrInfo;
+    return connect((int) fd, addr->ai_addr, (int) addr->ai_addrlen);
+}
+
+JNIEXPORT void JNICALL Java_io_questdb_network_Net_freeAddrInfo
+        (JNIEnv *e, jclass cl, jlong address) {
+    if (address != 0) {
+        freeaddrinfo((void *) address);
+    }
+}
+
+JNIEXPORT jlong JNICALL Java_io_questdb_network_Net_getAddrInfo
+        (JNIEnv *e, jclass cl, jlong host, jint port) {
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
```

Comment:
minor: isn't stack allocated value initialised with 0s in C?

Reply:
indeed, stack cannot be assumed as zeros